datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -2815,4 +2577,321 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_in_list_dictionary_int32() -> Result<()> {
+        // Create schema with dictionary-encoded Int32 column
+        let dict_type =
+            DataType::Dictionary(Box::new(DataType::Int8), Box::new(DataType::Int32));
+        let schema = Schema::new(vec![Field::new("a", dict_type.clone(), false)]);
+        let col_a = col("a", &schema)?;
+
+        // Create IN list with Int32 literals: (1, 2, 3)
+        let list = vec![lit(1i32), lit(2i32), lit(3i32)];
+
+        // Create InListExpr via in_list() - this uses Int32StaticFilter for Int32 lists
+        let expr = in_list(col_a, list, &false, &schema)?;
+
+        // Create dictionary-encoded batch with values [1, 2, 5]
+        // Dictionary: keys [0, 1, 2] -> values [1, 2, 5]
+        let keys = Int8Array::from(vec![0, 1, 2]);
+        let values = Int32Array::from(vec![1, 2, 5]);
```

Comment:
I personally recommend using values that are clearly not the keys - for example

```suggestion
        let values = Int32Array::from(vec![100, 200, 500]);
```

( you would also have to change the literals above)